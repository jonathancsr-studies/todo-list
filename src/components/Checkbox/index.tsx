import styles from "./styles.module.css";

interface CheckboxProps {
  value: boolean;
  onChange: () => void;
}
export function Checkbox({ value }: CheckboxProps) {
  return <input type="checkbox" className={styles.checkbox} checked={value} />;
}
