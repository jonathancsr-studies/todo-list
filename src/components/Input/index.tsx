import { InputHTMLAttributes } from "react";

type InputProps = InputHTMLAttributes<HTMLInputElement>;

import styles from "./styles.module.css";

export function Input({ ...props }: InputProps) {
  return <input className={styles.input} {...props} type="text" />;
}
