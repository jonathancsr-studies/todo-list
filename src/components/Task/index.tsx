export interface Task {
  name: string;
}

interface TaskProps {
  task: Task;
}

export function Task({ task }: TaskProps) {
  return (
    <span
      style={{
        margin: "0.25rem 0",
        padding: "1rem",
        backgroundColor: "#f5f5f5",
      }}
    >
      {task.name}
    </span>
  );
}
