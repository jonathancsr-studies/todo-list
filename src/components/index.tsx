export { Header } from "./Header";
export { Input } from "./Input";
export { Task } from "./Task";
export { TaskContainer } from "./TaskContainer";
