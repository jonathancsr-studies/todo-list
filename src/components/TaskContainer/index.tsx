import { Task } from "../Task";

interface TaskContainerProps {
  tasks: Task[];
}

export function TaskContainer({ tasks = [] }: TaskContainerProps) {
  return (
    <div
      style={{
        display: "flex",
        flexDirection: "column",
        justifyContent: "center",
        alignItems: "center",
      }}
    >
      {tasks.map((task) => (
        <Task key={task.name} task={task} />
      ))}
    </div>
  );
}
