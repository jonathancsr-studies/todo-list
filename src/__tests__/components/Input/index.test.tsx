import { render, screen } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { vi } from "vitest";
import { Input } from "../../../components";

describe("Testing Input", () => {
  const inputPlaceholder = "Escreva aqui";
  const inputName = "text-input";

  it("should be render text input successfully", () => {
    const { container } = render(
      <Input name={inputName} placeholder={inputPlaceholder} type="text" />
    );

    const inputElement = screen.getByRole("textbox");

    expect(inputElement).toBeInTheDocument();
    expect(container).toMatchSnapshot();
  });

  it("should be able to change value", async () => {
    const handleChange = vi.fn();
    const userText = "test value";
    render(
      <Input
        name={inputName}
        placeholder={inputPlaceholder}
        type="text"
        onChange={handleChange}
      />
    );
    const inputElement = screen.getByRole("textbox");

    await userEvent.type(inputElement, userText);

    expect(handleChange).toHaveBeenCalled();
    expect(inputElement).toHaveValue(userText);
  });
});
