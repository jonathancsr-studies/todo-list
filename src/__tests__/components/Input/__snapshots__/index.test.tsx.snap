// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Testing Input > should be render text input successfully 1`] = `
<div>
  <input
    name="text-input"
    placeholder="Escreva aqui"
    type="text"
  />
</div>
`;
